lib/caasp.pm

```diff
@@ -156,12 +156,29 @@ sub check_reboot_changes {
 
 # Install a pkg in Kubic
 sub trup_install {
+    my ($input, $unnecessary, $necessary, $pkg_is_present) = @_;
+    my ($pkg_required, $not_needed) = 0;
+
     # rebootmgr has to be turned off as prerequisity for this to work
     script_run "rebootmgrctl set-strategy off";
-    my $package = shift;
-    trup_call("pkg install $package");
-    process_reboot 1;
-    assert_script_run("rpm -qi $package");
+    my @pkg = split(' ', $input);
+    foreach (@pkg) {
+        if (!script_run("rpm -q $_")) {
+            $unnecessary .= join ' ', $_, ' ';
```

Comment:
Join is for making string from array, for example `$to_install = join(' ', @pkgs)`- I don't understand it here.
Maybe you wanted `$unnecessary .= "$_ ";` ?

Reply:
Let me make the changes and test again ;) Thanks